activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -90,6 +98,11 @@ def commit_records
         ite.each do |i|
           i.committed!(should_run_callbacks: false)
         end
+        @records.uniq.each do |i|
+          if i.weakref_alive?
+            i.committed!(should_run_callbacks: false)
+          end
```

Comment:
this is the state it restores https://github.com/rails/rails/blob/master/activerecord/lib/active_record/transactions.rb#L306